edisgo/network/topology.py

```diff
@@ -2399,61 +2455,199 @@ def connect_to_lv_based_on_geolocation(
+                "generator": self.add_generator,
+                "charging_point": self.add_load,
+                "heat_pump": self.add_load,
+                "storage_unit": self.add_storage_unit,
+            }
+            add_func = add_func_map.get(comp_type)
+            if add_func is None:
+                raise ValueError(
+                    f"Provided component type {comp_type} is not valid. Must either be "
+                    f"'generator', 'charging_point', 'heat_pump' or 'storage_unit'."
+                )
+            return add_func
+
+        def handle_voltage_level_6():
+            # get substations and MV buses
+            substations = self.buses_df.loc[self.transformers_df.bus1.unique()]
+            if allow_mv_connection:
+                mv_buses = self.buses_df.loc[self.mv_grid.buses_df.index]
+            else:
+                mv_buses = pd.DataFrame()
+            all_buses = pd.concat([substations, mv_buses])
+            # calculate distance to possible buses
+            target_bus, target_bus_distance = geo.find_nearest_bus(
+                geolocation, all_buses
+            )
+            if target_bus in substations.index:
+                # if distance is larger than allowed, create new bus and connect to
+                # station via a new line
+                if target_bus_distance > max_distance_from_target_bus:
+                    bus = self._connect_to_lv_bus(
+                        edisgo_object, target_bus, comp_type, comp_data
+                    )
+                else:
+                    mvlv_subst_id = self.buses_df.loc[target_bus].loc["lv_grid_id"]
+                    comp_data["mvlv_subst_id"] = mvlv_subst_id
+                    comp_name = self.connect_to_lv(edisgo_object, comp_data, comp_type)
+                    return None, comp_name
+            else:
+                bus = target_bus
+            return bus, None
+
+        def handle_voltage_level_7():
+            if allow_mv_connection:
+                mv_buses = self.buses_df.loc[self.mv_grid.buses_df.index]
+                mv_buses = geo.calculate_distance_to_buses_df(geolocation, mv_buses)
+                mv_buses_masked = mv_buses.loc[
+                    mv_buses.distance < max_distance_from_target_bus
+                ]
+
+            lv_buses = self.buses_df.drop(self.mv_grid.buses_df.index)
+            if comp_type == "charging_point":
+                if comp_data["sector"] == "home":
+                    lv_loads = self.loads_df[
+                        self.loads_df.sector.isin(["residential", "home"])
+                    ]
+                    lv_loads = lv_loads.loc[
+                        ~lv_loads.bus.isin(self.mv_grid.buses_df.index)
+                    ]
+                elif comp_data["sector"] == "work":
+                    lv_loads = self.loads_df[
+                        self.loads_df.sector.isin(
+                            ["industrial", "cts", "agricultural", "work"]
```

Comment:
Same here as above, I would not consider work, as these are charging points then and no businesses.